feat!: rename faas to function

zroubalik opened the pull request:
Signed-off-by: Zbynek Roubalik <zroubali@redhat.com>

This PR renames:
- binary: `faas` -> `function`
- env variables: `FAAS_` -> `FUNCTION_`
- config: `faas.yaml` -> `func.yaml`
- release registry: `quay.io/boson/faas` -> `quay.io/boson/function`

Fixes: #201

lance commented:
> release registry: quay.io/boson/faas -> quay.io/boson/faas

I assume this is a typo :smile:

lance commented on README.md:
```diff
@@ -9,7 +9,7 @@
 [Demo Screencast]
 
-faas is a "Function as a Service" Client Library and CLI for enabling the development of implicitly deployed, platform agnostic code.
+function is a Client Library and CLI for enabling the development of implicitly deployed, platform agnostic code.
```
```suggestion
`function` is a Client Library and CLI for enabling the development of implicitly deployed, platform agnostic code.
```
Should the CLI binary be back-ticked?

lance commented on cmd/root.go:
```diff
@@ -17,13 +17,13 @@ import (
+	Long: `Mangage function projects.
 
-Create and run Functions as a Service.`,
+	Create, build and deploy functions as Knative Services.`,
```
Should this be flush left so that it's not indented when run by the user?

lance commented on docs/README.md:
```diff
@@ -1,8 +1,8 @@
 [Demo Screencast]
 
-`faas` is a "Function as a Service" Client Library and CLI for enabling the development of implicitly deployed, platform agnostic code.
+`function` is a Client Library and CLI for enabling the development of implicitly deployed, platform agnostic code.
```
Ah! It is back-ticked here. :)

zroubalik replied:
you don't want it to be back-ticked?

zroubalik replied:
gotcha :)

zroubalik commented:
> > release registry: quay.io/boson/faas -> quay.io/boson/faas
> 
> I assume this is a typo 😄

Yeah :) updated the description

lance commented on templates/go/http/README.md:
```diff
@@ -8,7 +8,7 @@ Welcome to your new Go Function! The boilerplate function code can be found in
 each feature, and confirm it works with `go test`.
 
-Update the running analog of the funciton using the `faas` CLI or client
+Update the running analog of the funciton using the `function` CLI or client
```
```suggestion
Update the running analog of the function using the `function` CLI or client
```

lance commented on templates/node/events/package.json:
```diff
@@ -13,9 +13,9 @@
+    "cloudevents": "^3.1.0",
     "faas-js-runtime": "0.4.0",
     "nodemon": "^2.0.4",
-    "cloudevents": "^3.1.0",
```
This change won't get picked up without running `pkger` and will conflict with https://github.com/boson-project/faas/pull/202

lance commented on templates/node/events/test/integration.js:
```diff
@@ -21,7 +21,7 @@ test('Integration: handles a valid event', t => {
       .set(Spec.id, 'TEST-EVENT-1')
       .set(Spec.source, 'http://localhost:8080/integration-test')
-      .set(Spec.type, 'dev.faas.example')
+      .set(Spec.type, 'dev.function.example')
```
Same as above. I should land https://github.com/boson-project/faas/pull/202, then if you can rebase on top of that and re-run pkger this should all be good.

lance reviewed (commented):
Looks good with a few small changes. The big thing is that I should land https://github.com/boson-project/faas/pull/202 and you should rebase, then do `pkger`. The Node.js and Quarkus changes you have made here won't get picked up without it. :)

zroubalik commented:
@lance thanks for the review, fixed

lance commented:
@zroubalik lgtm - unless @rhuss has any concerns let's do it!

rhuss commented on docs/getting_started_kubernetes.md:
````diff
@@ -55,7 +55,7 @@ First edit `knative/config-domain.yaml` to contain your domain of choice and the
 kubectl apply -f knative/config-domain.yaml
 ```
-Note that this step is [pending automation](https://github.com/boson-project/faas/issues/47)
+Note that this step is [pending automation](https://github.com/boson-project/function/issues/47)
````
don't think this is correct.

rhuss commented on cmd/root.go:
```diff
@@ -17,13 +17,13 @@ import (
 	SilenceErrors: true, // we explicitly handle errors in Execute()
 	SilenceUsage:  true, // no usage dump on error
-	Long: `Function as a Service
+	Long: `Mangage function projects.
```
```suggestion
	Long: `Manage function projects.
```

rhuss commented on docs/README.md:
```diff
@@ -1,8 +1,8 @@
-# FAAS Documentation
+# FUNCTION Documentation
```
```suggestion
# Function Documentation
```

rhuss commented on config.go:
```diff
@@ -9,7 +9,7 @@ import (
 
 // ConfigFile is the name of the config's serialized form.
-const ConfigFile = "faas.yaml"
+const ConfigFile = "function.yaml"
```
Wonder whether a shorter file name (`func.yaml`) makes sense ? If we e.g. add `func` as alias this would be a natural naming, too.

zroubalik replied:
yeah, this makes sense. I'll modify it

rhuss commented on templates/node/events/package-lock.json:
```diff
@@ -1,8 +1,2837 @@
 {
   "name": "event-handler",
   "version": "0.1.0",
-  "lockfileVersion": 1,
```
Is this file supposed to be part of this renaming PR ?

zroubalik replied:
hm, don't know how does this end up here.

zroubalik replied:
these files get modified after running make test. The deps needed to be updated.

rhuss reviewed (commented):
Looks mostly good to me (but see my comments).

Alson not sure about that pkged.go file (can't add a comment there). What is this ?

Beside this (and reading the changes), I wonder whether we should add at least `func` as an alias as this is much shorter and would be a perfect drop-in replacement for `faas` when it comes to the length of the command. Also I would be in favor of a `func.yaml` over `function.yaml`.

zroubalik commented:
@rhuss Thanks for the review. I have fixed the things from your comments, and renamed `faas.yaml` -> `func.yaml`.
func alias would be nice, but we can cover it in the separate PR imho, if we agree on it.

Is it ok to merge?

rhuss commented on docs/commands.md:
```diff
@@ -7,116 +7,116 @@ Creates a new Function project at _`path`_. If _`path`_ is unspecified, assumes
 ## `delete`
 
-Removes a deployed function from the cluster. The user may specify a function by name, path or if neither of those are provided, the current directory will be searched for a `faas.yaml` configuration file to determine the function to be removed. The namespace defaults to the value in `faas.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be specifi ...
+Removes a deployed function from the cluster. The user may specify a function by name, path or if neither of those are provided, the current directory will be searched for a `function.yaml` configuration file to determine the function to be removed. The namespace defaults to the value in `function.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be ...
```
function.yaml -> func.yaml (I know this file is generated, but you got the idea ;-).

rhuss commented on docs/commands.md:
```diff
@@ -7,116 +7,116 @@ Creates a new Function project at _`path`_. If _`path`_ is unspecified, assumes
 ## `list`
 
-Lists all deployed functions. The namespace defaults to the value in `faas.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace defaults to the value in `faas.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be specified on the command line, and if so this will overwrite the value in `faas.yaml`.
+Lists all deployed functions. The namespace defaults to the value in `function.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace defaults to the value in `function.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be specified on the command line, and if so this will overwrite the value in `function.yaml`.
```
function.yaml -> func.yaml

rhuss commented on docs/commands.md:
```diff
@@ -7,116 +7,116 @@ Creates a new Function project at _`path`_. If _`path`_ is unspecified, assumes
 ## `describe`
 
-Prints the name, route and any event subscriptions for a deployed Function. The user may also specify the name of the function to describe. The namespace defaults to the value in `faas.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be specified on the command line, and if so this will overwrite the value in `faas.yaml`.
+Prints the name, route and any event subscriptions for a deployed Function. The user may also specify the name of the function to describe. The namespace defaults to the value in `function.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be specified on the command line, and if so this will overwrite the value in `function.yaml`.
```
function.yaml -> func.yaml

rhuss reviewed (commented):
looks good but there are still some references to function.yaml

rhuss commented:
Thanks !

/lgtm
/approve
